File: app/src/main/java/com/example/waveplayer/ViewModelUserPickedPlaylist.java
package com.example.waveplayer;

import androidx.annotation.GuardedBy;
import androidx.lifecycle.ViewModel;

import com.example.waveplayer.random_playlist.RandomPlaylist;

public class ViewModelUserPickedPlaylist extends ViewModel {

    @GuardedBy("this")
    volatile private RandomPlaylist userPickedPlaylist;

    synchronized public RandomPlaylist getUserPickedPlaylist(){
        return userPickedPlaylist;
    }

    synchronized public void setUserPickedPlaylist(RandomPlaylist userPickedPlaylist) {
        this.userPickedPlaylist = userPickedPlaylist;
    }

}

File: app/src/main/java/com/example/waveplayer/ViewModelUserPickedPlaylist.kt
package com.example.waveplayer

import androidx.annotation.GuardedBy
import androidx.lifecycle.ViewModel
import com.example.waveplayer.random_playlist.RandomPlaylist

class ViewModelUserPickedPlaylist constructor() : ViewModel() {
    @GuardedBy("this")
    @Volatile
    private var userPickedPlaylist: RandomPlaylist? = null
    @Synchronized
    fun getUserPickedPlaylist(): RandomPlaylist? {
        return userPickedPlaylist
    }

    @Synchronized
    fun setUserPickedPlaylist(userPickedPlaylist: RandomPlaylist?) {
        this.userPickedPlaylist = userPickedPlaylist
    }
}